.github/workflows/build-pr.yml

```diff
@@ -0,0 +1,64 @@
+name: Build-PR-test
+
+on:
+  # run in a read-write copy of the target of the PR (CombatExtended-Continued/CombatExtended/Development)
+  pull_request_target: 
+    branches: [ Development ]
+
+jobs:
+  build:
+    # Could use windows with dotnet, but I don't know the tooling.
+    # Also, mono is the less advanced compiler, so if it builds successfully here, it'll work anywhere.
+    runs-on: ubuntu-latest 
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: prepare rebuild
+        run: |
+          # Set up who gets blamed for the added assembly
+          git config --global user.email "actions@github.com"
+          git config --global user.name "GitHub Actions"
+
+          # Grab the code from the PR, using the version from the PR in case of conflicts.
+          git pull --allow-unrelated-histories -X theirs origin ${{ github.event.pull_request.head.sha }}:
+
+          # Grab the RimWorld reference assemblies from nuget
+          wget https://www.nuget.org/api/v2/package/Krafs.Rimworld.Ref/1.2.2753 -O rwref.zip
+          unzip rwref.zip -d reference
+
+          # We've checked out code from a PR, we can't trust it (it could dump environment variables or otherwise do unapproved things)
+          # So we grab the build script from the trusted repo.  It does mean changes to the build script itself won't get auto-tested,
+          # but that's unavoidable in a PR.
+          wget https://raw.githubusercontent.com/CombatExtended-Continued/CombatExtended/Development/Make.py -O - > Make.py
+
+      # It would be better to only create the destination branch once we know the build will succeed, but we have to commit the pull from line 23 or it chokes.
+      - uses: peterjgrainger/action-create-branch@v2.0.1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          branch: test-branch-${{ github.event.pull_request.head.ref }}-${{ github.run_number }}
```

Comment:
Does this means it creates a new testing branch for every push to the PR branch?

Reply:
Yes.  I meant to list this as a decision point, as there are pros and cons to this approach vs. creating and then updating a single testing branch.  

First, it's simpler.  The branch is guaranteed a unique name, so there's no need to check if the branch exists before creating it.  Nor is there a need to sync the testing branch despite whatever has been done to the source's history.  

Second, it creates "breadcrumbs" for how the PR got to its current state.  If someone wants to compare a previous state of the PR, it's easy to do so.  We could do the same by just committing to the same branch, but that prevents solving problem one by `git reset --hard`.

There are two downsides.  First, it creates additional branch clutter to be deleted.  This is fairly minor, as bulk branch actions are pretty straightforward.  Second, it means a tester can't just fast forward their local branch to test the latest changes, they have to switch branches instead.

As another consideration, the branch names are guaranteed to be unique between PRs in addition to within PRs.  This means if a future PR is created with the same name, before the old test branches have been cleaned up, this handles that.  It could be done by using `test-branch-<pr-name>-<pr-number>` or similar, so that's not a *major* consideration.

I'm not sure which approach is better, but did this because it was simpler to implement.